docs/src/userguide/memoryviews.rst

```diff
@@ -52,27 +63,70 @@ Syntax
 
 Memory views use Python slicing syntax in a similar way as NumPy.
 
-To create a complete view on a one-dimensional int buffer::
+To create a complete view on a one-dimensional ``int`` buffer:
+
+.. tabs::
+
+    .. group-tab:: Pure Python
+
+        .. code-block:: python
+
+            view1D: cython.int[:] = exporting_object
+
+    .. group-tab:: Cython
+
+        .. code-block:: cython
+
+            cdef cython.int[:] view1D = exporting_object
 
-    cdef int[:] view1D = exporting_object
+A complete 3D view:
 
-A complete 3D view::
+.. tabs::
 
-    cdef int[:,:,:] view3D = exporting_object
+    .. group-tab:: Pure Python
+
+        .. code-block:: python
+
+            view3D: cython.int[:,:,:] = exporting_object
+
+    .. group-tab:: Cython
+
+        .. code-block:: cython
+
+            cdef int[:,:,:] view3D = exporting_object
 
 They also work conveniently as function arguments:
 
-.. code-block:: cython
+.. tabs::
+
+    .. group-tab:: Pure Python
 
-    def process_3d_buffer(int[:,:,:] view not None):
-        ...
+        .. code-block:: python
+
+            def process_3d_buffer(view: cython.int[:,:,:]):
+                ...
+
+    .. group-tab:: Cython
+
+        .. code-block:: cython
+
+            def process_3d_buffer(int[:,:,:] view not None):
+                ...
 
 The ``not None`` declaration for the argument automatically rejects
-None values as input, which would otherwise be allowed.  The reason why
-None is allowed by default is that it is conveniently used for return
+``None`` values as input, which would otherwise be allowed.  The reason why
+``None`` is allowed by default is that it is conveniently used for return
 arguments:
```

Comment:
There is no `not None` declaration in the Python variant. And it's also not necessary because we reject `None` as memoryview input.

Reply:
OK I have slightly extended this part to include pure python mode.